hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/HoodieAppendHandle.java

```diff
@@ -512,6 +512,7 @@ public List<WriteStatus> close() {
         status.getStat().setFileSizeInBytes(logFileSize);
       }
 
+      createCompletedMarkerFile(partitionPath, baseInstantTime);
```

Comment:
Can we create the file only if necessary, when `enforceFinalizeWriteCheck()` or `enforceCompletionMarkerCheck()` is true.

Reply:
Will update the diff, after adding this check.  (We have this enabled by default, it make sense to wrap it up with the flag for OSS).

Reply:
Done. Added the check and changed the function to createCompletedMarkerFileIfRequired()